File: Array/Remove Duplicates/removeDuplicates.js
function removeDuplicates(nums) {
  if (nums.length === 0) return 0; // Empty array, nothing to remove.
  
  let uniqueCount = 1; // At least the first element is unique.
    
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== nums[i - 1]) {
      nums[uniqueCount] = nums[i];
      uniqueCount++;
    }
  }
  
  nums.length = uniqueCount; // Truncate the array to remove duplicates.
  return uniqueCount; // Return the length of the new array without duplicates.
}

console.log(removeDuplicates([1, 1, 2, 2, 2, 3, 4, 4, 5])); //5
console.log(removeDuplicates([0, 1, 2, 1])); //4

File: 26-Remove Duplicates from Sorted Array/RemoveDuplicates.java
// File: Solution.java

class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0; // If the array is empty, return 0
        
        int k = 1; // Initialize k as the pointer for the position of the next unique element
        for (int i = 1; i < nums.length; i++) {
            
            // Check if the previous element and the current element are not the same
            if (nums[i] != nums[i - 1]) {
                nums[k] = nums[i]; // Place the current element at the k-th position
                k++; // Increment k to point to the next unique position
            }
        }
        return k; // Return the number of unique elements
    }
}

// File: RemoveDuplicates.java

public class RemoveDuplicates {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        int k = solution.removeDuplicates(nums);
        System.out.println("Number of unique elements: " + k);
        System.out.print("Array after removing duplicates: ");
        for (int i = 0; i < k; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}
